sympy/calculus/finite_diff.py

```diff
@@ -387,6 +395,9 @@ def _as_finite_diff(derivative, points=1, x0=None, wrt=None):
         x0 = wrt
 
     if not iterable(points):
+        if hasattr(points, 'is_Function'):
+            if points.is_Function and wrt in points.args:
+                points = points.subs(wrt, x0)
```

Comment:
is_Function should be defined for all Basic instances. Why do we need to check if the attribute is present. Also you can do this more efficiently and cleanly with
```
if getattr(points, 'is_Function', False) and wrt in points.args:
```

Reply:
>  Why do we need to check if the attribute is present?

I was worried that some users might set ``points`` to be Python integer (or float) and not Sympy, which would results in an unwanted error. By checking if it has ``is_Function`` the code does not break. You are right using `getattr` is a better idea, thanks.

What would be the best way to check if ``points`` is an object like `Function('dx')(Symbol('x')`?

Reply:
```python
>>> isinstance(f(x), Function) and len(f(x).args) == 1
True
```
That will also pass if `f` is `exp`.